scripts/nbconverted/main_Pa_sim_simple_linear.py

```diff
@@ -0,0 +1,331 @@
+    analysis_dir = os.path.join(each_dir, analysis_name)
+    if os.path.exists(analysis_dir):
+        print('directory already exists: {}'.format(analysis_dir))
+    else:
+        os.mkdir(analysis_dir)
+        print('creating new directory: {}'.format(analysis_dir))
+
+
+# In[20]:
+
+
+# Output the new gene expression values for each sample
+train_input_file = os.path.join(base_dirs[0], analysis_name, "train_model_input.txt.xz")
+data_holdout.to_csv(train_input_file, sep='\t', compression='xz', float_format="%.5g")
+
+# Output log file with params
+log_file = os.path.join(os.path.dirname(os.getcwd()), 'metadata', analysis_name)
+
+args_dict = {
+    "gene A": geneA,
+    "size of gene set C": geneSetC_size,
+    "genes in set C": geneSetC,
+    "proportion of gene A to use to update genes in C": proportion,
+    "threshold of A activation": thresholdA,
+    "effect size of genes in set C": effect_sizeA,
+}
+
+with open(log_file, 'w') as f:
+    for key, value in args_dict.items():
+        f.write('%s: %s\n' % (key, value))
+        
+# Output geneA only file
+geneA_file = os.path.join(base_dirs[0], analysis_name, geneA + ".txt")
+geneA_only.to_csv(geneA_file, sep='\t', float_format="%.5g")
+
+
+# In[21]:
+
+
+get_ipython().run_line_magic('time', '')
+# Train models on input dataset
+
+# params
+learning_rate = 0.001
+batch_size = 100
+epochs = 200
+kappa = 0.01
+intermediate_dim = 100
+latent_dim = 10
+epsilon_std = 1.0
+num_PCs = latent_dim
+
+base_dir = os.path.dirname(os.getcwd())
+vae.tybalt_2layer_model(learning_rate, batch_size, epochs, kappa, intermediate_dim, latent_dim, epsilon_std, base_dir, analysis_name)
+pca.pca_model(base_dir, analysis_name, num_PCs)
+
+
+# Define offset vectors in gene space
+data_dir = os.path.join(base_dirs[0], analysis_name)
+target_gene = geneA
+percent_low = 5
+percent_high = 95
+
+def_offset.gene_space_offset(data_dir, target_gene, percent_low, percent_high)
+
+
+# Define offset vectors for different latent spaces
+model_dir = os.path.join(base_dirs[2], analysis_name)
+encoded_dir = os.path.join(base_dirs[1], analysis_name)
+
+def_offset.vae_latent_space_offset(data_dir, model_dir, encoded_dir, target_gene, percent_low, percent_high)
+def_offset.pca_latent_space_offset(data_dir, model_dir, encoded_dir, target_gene, percent_low, percent_high)
+
+
+# Predict gene expression using offset in gene space and latent space
+out_dir = os.path.join(base_dirs[3], analysis_name)
+
+interpolate.interpolate_in_gene_space(data_dir, target_gene, out_dir, percent_low, percent_high)
+interpolate.interpolate_in_vae_latent_space(data_dir, model_dir, encoded_dir, target_gene, out_dir, percent_low, percent_high)
+interpolate.interpolate_in_pca_latent_space(data_dir, model_dir, encoded_dir, target_gene, out_dir, percent_low, percent_high)
+
+
+# Plot prediction per sample along gradient of PA1673 expression
+viz_dir = os.path.join(base_dirs[5], analysis_name)
+plot.plot_corr_gradient(out_dir, viz_dir)
+
```

Comment:
Also, the notebook is printing some strangely formatted output to the screen. Probably want to check that out

Reply:
This output looks to be just a traceback of the calls that python made when running the script so its fine.

Reply:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-21-862bdeb8f6f9> in <module>()
     45 # Plot prediction per sample along gradient of PA1673 expression
     46 viz_dir = os.path.join(base_dirs[5], analysis_name)
---> 47 plot.plot_corr_gradient(out_dir, viz_dir)

~/Documents/Repos/Pseudomonas_latent_spaces/scripts/functions/plot.py in plot_corr_gradient(out_dir, viz_dir)
     74     fg=sns.jointplot(x='gene_space', y='latent_space_vae', data=X, kind='hex');
     75     fg.set_axis_labels('Gene space correlation', 'Latent space VAE correlation')
---> 76     fg.set_title('Gene space vs Latent VAE space')
     77     fg.savefig(fig2_file)
     78 

AttributeError: 'JointGrid' object has no attribute 'set_title'
```
Is it that? That looks like an error to me.